posthog/storage/object_storage.py

```diff
@@ -30,3 +33,19 @@ def read(file_name: str):
     s3_object = s3.Object("posthog", file_name)
     content = s3_object.get()["Body"].read()
     return content.decode("utf-8")
+
+
+def list_all():
+    # TODO page these instead of loading them all
+    return s3.Bucket("posthog").objects.all()
+
+
+def delete(older_than: datetime.datetime):
+    date_limit = pytz.UTC.localize(older_than)
+    count = 0
+    for object in list_all():
```

Comment:
Nit: a possible scaling issue of doing list_all -> `s3.Bucket("posthog").objects.all()` is that the list could be technically unbounded and return a lot of data. Not sure if the SDK supports it but if yes we could technically something like `s3.Bucket("posthog").objects.filter(last_modified<date_limit).all()`